keymap/vim.js

```diff
@@ -247,15 +247,23 @@
     }
 
     function detachVimMap(cm, next) {
-      if (this == CodeMirror.keyMap.vim)
+      if (this == CodeMirror.keyMap.vim) {
         CodeMirror.rmClass(cm.getWrapperElement(), "cm-fat-cursor");
+        if(cm.state.matchBrackets) {
+          cm.state.matchBrackets.fatCursor = false;
```

Comment:
Mutating a random object that isn't owned by this code isn't a good idea. Doesn't the code that checks for the existance of a fat cursor in the dom already handle this?